### PR TITLE
C++: hide internal types

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

File: api/sixtyfps-cpp/include/sixtyfps.h [hunk 1]
```diff
@@ -477,7 +477,7 @@ class VectorModel : public Model<ModelData>
 template<typename C, typename ModelData>
 class Repeater
```
Comment (ogoffart) [inline api/sixtyfps-cpp/include/sixtyfps.h:478]: I guess the Repeater itself need to be moved.

File: xtask/src/cbindgen.rs [hunk 2]
```diff
@@ -308,7 +308,7 @@ fn gen_corelib(root_dir: &Path, include_dir: &Path) -> anyhow::Result<()> {
             r"
 namespace sixtyfps {{
     namespace private_api {{ enum class VersionCheck {{ Major = {}, Minor = {}, Patch = {} }}; class ComponentWindow; }}
-    namespace cbindgen_private {{ using sixtyfps::private_api::ComponentWindow; using namespace vtable; struct KeyEvent; }}
+    namespace cbindgen_private {{ using sixtyfps::private_api::ComponentWindow; using namespace vtable; struct KeyEvent; using namespace private_api; }}
```
Comment (ogoffart) [inline xtask/src/cbindgen.rs:311]: what's the reason for this change?

**Review by ogoffart — APPROVED** [review]

File: xtask/src/cbindgen.rs (hunk 2 above)
Comment (tronical) [reply xtask/src/cbindgen.rs:311]: For the structs for different items, cbindgen maps `width: Property<f32>` to `Property<f32> width`. Since the generated structs live in `cbindgen_private` this was an easy way for that to compile. If you prefer I can also look for an alternate way so that cbindgen generates `private_api::Property<f32>` instead.

File: api/sixtyfps-cpp/include/sixtyfps.h (hunk 1 above)
Comment (tronical) [reply api/sixtyfps-cpp/include/sixtyfps.h:481]: Done, along with more types.

File: xtask/src/cbindgen.rs (hunk 2 above)
Comment (ogoffart) [reply xtask/src/cbindgen.rs:311]: I guess it's fine like that.

Maybe one want to be a bit more explicit and only do `using private_api::Property;`

**Review by ogoffart — APPROVED** [review]

File: xtask/src/cbindgen.rs (hunk 2 above)
Comment (tronical) [reply xtask/src/cbindgen.rs]: Done
